schema/post.graphql

```diff
@@ -0,0 +1,42 @@
+type Post {
+    id: ID!
+    authorID: ID!
+    content: String!
+    createdAt: Int
+    updatedAt: Int
+    deletedAt: Int
+}
+
+input PostFilter {
+    search: String
+    where: PostWhere
+}
+
+input PostWhere {
+    id: IDFilter
+    authorID: IDFilter
+    lastName: StringFilter
+    content: StringFilter
+    createdAt: IntFilter
+    updatedAt: IntFilter
+    deletedAt: IntFilter
+}
+
+input PostCreateInput {
+    authorID: ID!
+    content: String!
+}
+
+input PostUpdateInput{
+    id: ID!
+    content: String!
+}
+
+input PostDeleteInput {
+    id: ID!
+}
+
+type PostsPayload {
+    posts: [Post!]!
```

Comment:
If an author has not created any post then if we query allPostByAuthor then we will receive an empty posts array.

Reply:
My understanding was that it would allow `[] (empty array)` as it is just non nullable array of non nullable Post, so empty array wouldn't have any Post and also empty array is not null

Reply:
[Post!]! means that at least one post will be present in the array.
[Post]! means that the array will contain 0 or more posts.

Reply:
According to [This GraphQL doc](https://graphql.org/learn/schema/#lists-and-non-null)

[Post!]  => [] is valid
and for [Post]! => [] is valid

I believe the `!` operator does not enforce the number of elements in the list, it simply dictates the non-nullability of the list,

Reply:
Yes. The above is correct
But we are discussing about [Post!]!
In this case it means that posts array with at least contain one post
But for authors that do not have any post we should receive an empty array

Reply:
Ok, the we should go with `[Post!]`,  for authors with no post it will return `null` or `empty array` which is allowed and none of the post will be null inside the array